django/contrib/admin/views/main.py

```diff
@@ -307,30 +290,16 @@ def get_results(self, request):
             full_result_count = self.root_queryset.count()
         else:
             full_result_count = None
-        can_show_all = result_count <= self.list_max_show_all
-        multi_page = result_count > self.list_per_page
-
-        # Get the list of objects to display on this page.
-        if (self.show_all and can_show_all) or not multi_page:
-            result_list = self.queryset._clone()
-        else:
-            try:
-                result_list = paginator.page(self.page_num).object_list
-            except InvalidPage:
-                raise IncorrectLookupParameters
-
-        self.result_count = result_count
         self.show_full_result_count = self.model_admin.show_full_result_count
         # Admin actions are shown if there is at least one entry
         # or if entries are not counted because show_full_result_count is disabled
         self.show_admin_actions = not self.show_full_result_count or bool(
             full_result_count
         )
         self.full_result_count = full_result_count
-        self.result_list = result_list
-        self.can_show_all = can_show_all
```

Comment:
The API of `get_results` is being changed here.
If this is the final approach, I don't see an issue with the change itself but we need to deprecate this properly:
ie:
- deprecation warning
- a workaround to populate the old properties while the deprecation is active